py/stencila/schema/util.py

```diff
@@ -0,0 +1,36 @@
+"""Utility functions for schema to/from JSON."""
+
+import json
+import typing
+
+from . import types
+from .types import Node, Entity
+
+
+def to_dict(node: Entity) -> dict:
+    """Convert an Entity node to a dictionary"""
+    node_dict = {
+        "type": node.__class__.__name__
+    }
+    node_dict.update(node.__dict__)
+    return node_dict
+
+
+def from_dict(node_dict: dict) -> typing.Union[dict, Node]:
```

Comment:
This does not handle child entities. It should traverse all descendant nodes and convert them too. See how we do it for R (where a `list` is the similar to a Python `dict`):

https://github.com/stencila/schema/blob/589adcff6b3c5db8c86a9c4a09b46db12f35278c/r/R/entity.R#L30-L31

Generally, we might save ourselves time by structuring the code similarly across the packages for the different languages.